internal/gatewayapi/securitypolicy.go

```diff
@@ -563,6 +563,12 @@ func (t *Translator) buildCORS(cors *egv1a1.CORS) *ir.CORS {
 		}
 	}
 
+	for _, origin := range cors.AllowRegexOrigins {
+		allowOrigins = append(allowOrigins, &ir.StringMatch{
+			SafeRegex: (*string)(&origin),
```

Comment:
is there any way to make sure the regex is valid?

Reply:
Hm, I'll look for validator.

Reply:
We can call regex.Validate(&origin) to validate the regex.